src/main/java/com/medicalclinic/service/PatientService.java

```diff
@@ -0,0 +1,37 @@
+package com.medicalclinic.service;
+
+import com.medicalclinic.exception.ProcessingPatientException;
+import com.medicalclinic.model.Patient;
+import com.medicalclinic.repository.PatientRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PatientService {
+    private final PatientRepository patientRepository;
+
+    public PatientService(PatientRepository patientRepository) {
+        this.patientRepository = patientRepository;
+    }
+
+    public List<Patient> getPatients() {
+        return patientRepository.listAll();
+    }
+
+    public Patient getPatientByEmail(String email) {
+        return patientRepository.findPatientByEmail(email).orElseThrow(() -> new ProcessingPatientException("Patient with email " + email + " not found"));
```

Comment:
return patientRepository.findPatientByEmail(email)
.orElseThrow(() -> new ProcessingPatientException("Patient with email " + email + " not found"));
do nowej linijki mozesz dac orElseThrow